lls/main.c

```diff
@@ -64,9 +116,24 @@ process_pkts(__attribute__((unused)) struct lls_config *lls_conf,
 			goto free_buf;
 
 		ether_type = eth_hdr->ether_type;
+		/*
+		 * Calculate offset to skip over VLAN header, if present.
+		 * TODO If VLAN stripping is enabled, we can skip this.
+		 */
+		ether_type = eth_hdr->ether_type;
+		next_hdr = (void *)((uint8_t *)(eth_hdr + 1) +
```

Comment:
This code is assuming that packets are always well formed. While it's okay to assume that a packet always has an Ethernet header, anything above could be missing. Unless you check the length, you may be accessing invalid memory positions.